File: src/Altinn.Platform/Altinn.Platform.PDF/src/main/java/altinn/platform/pdf/models/LayoutSettings.java
package altinn.platform.pdf.models;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;

@Schema(description = "Layout settings")
@Getter
@Setter
public class LayoutSettings {
  private PagesSettings pages;
  private ComponentSettings components;
}

File: src/App/backend/src/Altinn.App.Core/Models/LayoutSettings.cs
using System.Text.Json.Serialization;
using Newtonsoft.Json;

namespace Altinn.App.Core.Models;

/// <summary>
/// Defines the layout settings
/// </summary>
public class LayoutSettings
{
    /// <summary>
    /// Optional JSON schema URI for layout settings.
    /// </summary>
    [JsonProperty(PropertyName = "$schema")]
    [JsonPropertyName("$schema")]
    public string? Schema { get; set; }

    /// <summary>
    /// Default data type for the layout folder.
    /// </summary>
    public string? DefaultDataType { get; set; }

    /// <summary>
    /// Pages
    /// </summary>
    public Pages? Pages { get; set; }

    /// <summary>
    /// Components
    /// </summary>
    public Components? Components { get; set; }
}
